lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -121,39 +122,77 @@ public function convertException($message, DriverException $exception)
         return new Exception\DriverException($message, $exception);
     }
 
+
     /**
      * {@inheritdoc}
      */
     public function createDatabasePlatformForVersion($version)
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
-            throw DBALException::invalidPlatformVersionSpecified(
-                $version,
-                '<major_version>.<minor_version>.<patch_version>'
-            );
-        }
+
+        // Detect vendor and version
 
         if (false !== stripos($version, 'mariadb')) {
-            return $this->getDatabasePlatform();
-        }
+            // MariaDB does not follow one rule for versions ;)
 
-        $majorVersion = $versionParts['major'];
-        $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
-        $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
 
-        if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
-            $patchVersion = '9';
-        }
 
-        $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
+            // Will catch things like '10.2.8-MariaDB-1~lenny-log'
+            preg_match('/(?P<major>\d{1,2})\.(?P<minor>\d{1,2})\.(?P<patch>\d+)/', $version, $firstMatch);
+            // Will catch things like '5.5.5-10.2.8-MariaDB-10.2.8+maria~xenial-log'
+            preg_match('/\-(?P<major>10)\.(?P<minor>\d{1,2})\.(?P<patch>\d+)/', $version, $secondMatch);
+
+            if (empty($firstMatch) && empty($secondMatch)
+            ) {
+                throw DBALException::invalidPlatformVersionSpecified(
+                    $version,
+                    'MariaDB-<major_version>.<minor_version>.<patch_version>' .
+                                 'or <major_version>.<minor_version>.<patch_version>-MariaDB'
+                );
+            }
+
+            $versionParts = array_merge($firstMatch, $secondMatch);
+
+            $majorVersion = $versionParts['major'];
+            $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
+            $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
+
+            $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
+            if (version_compare($version, '10.2.7', '>=')) {
+                return new MariaDb102Platform();
```

Comment:
Would it make sense to name the class `MariaDb1027Platform`? Otherwise, it would be confusing that a connection to MariaDB 10.2.6 doesn't use the `MariaDb102Platform` class.

Reply:
Mmmmh, seems too much for me. Fix was released in 10.2.7, the first stable (GA) was 10.2.6. Will be forgotten soon. I guess not distributions include the first stable, and 10.2.6 was too much buggy for people to not update.